blockchain/filedao/filedao_v2_manager.go

```diff
@@ -90,6 +90,24 @@ func (fm *FileV2Manager) FileDAOByHeight(height uint64) FileDAO {
 	return nil
 }
 
+// AddFileDAO add a new v2 file
+func (fm *FileV2Manager) AddFileDAO(fd *fileDAOv2, start uint64) error {
+	// update current top's end
+	top := fm.Indices[len(fm.Indices)-1]
+	end, err := top.fd.Height()
+	if err != nil {
+		return err
+	}
+	if start != end+1 {
+		// new file's first block != current tip height
+		return ErrInvalidTipHeight
+	}
+
+	top.end = end
+	fm.Indices = append(fm.Indices, &fileV2Index{fd: fd, start: start, end: end})
```

Comment:
end is set to start -1 now

Reply:
yes, `end` is never used, until being updated next time a new file is added

see at beginning of
```go
func (fm *FileV2Manager) FileDAOByHeight(height uint64) FileDAO {
	right := len(fm.Indices) - 1
	if height >= fm.Indices[right].start {
		return fm.Indices[right].fd
	}
```
if input height >= [right].start, it knows height belongs to top fd and returns, `[right].end` is not used for comparison